core/common/src/main/java/alluxio/underfs/UnderFileSystemConfiguration.java

```diff
@@ -127,7 +131,7 @@ public UnderFileSystemConfiguration setShared(boolean shared) {
    * @return the updated configuration object
    */
   public UnderFileSystemConfiguration setUserSpecifiedConf(Map<String, String> ufsConf) {
-    mUfsConf = ufsConf;
+    mUfsConf.merge(ufsConf, Source.UFS_OPTION);
```

Comment:
this is a little different from before - it might leave behind some key/value pairs from the previous conf. Is this intentional?